source/daplink/interface/swd_host.c

```diff
@@ -894,6 +898,11 @@ uint8_t swd_set_target_state_hw(TARGET_RESET_STATE state)
             os_dly_wait(2);
             swd_set_target_reset(0);
             os_dly_wait(2);
+            
+            // Enable debug again because TARGET RESET might have disabled it
+            if (!swd_write_word(DBG_HCSR, DBGKEY | C_DEBUGEN)) {
+                return 0;
+            }
```

Comment:
Well, if that function had been in use, which its not currently, we would need to enable debug mode again after calling the reset function, because we disable debug in the reset function now.